tests/functional/test_issue-456/main.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  
+  default = {
+    repository_1 = {
+      name                 = "bar1"
+      image_tag_mutability = "MUTABLE"
+
+      scan_on_push = true
+    },
+    repository_2 = {
+      name                 = "bar2"
+      image_tag_mutability = "MUTABLE"
+
+      scan_on_push = true
+    },
+    repository_1 = {
+      name                 = "bar3"
+      image_tag_mutability = "MUTABLE"
+
+      scan_on_push = false
+    }
+
+  }
+}
+
+
+
+module "ecr_repository" {
+  source   = "./modules/ecr_module"
+  for_each = var.ecr_repositories
+
+  name          = each.key
+  scan_on_push  = each.value.scan_on_push
+}
+
+module "other_ecr_repository" {
+  source   = "./modules/other_ecr_module"
+
+  name                 = "non_for_loop_bar"
+  scan_on_push = true
+}
```

Comment:
can we also have a test module with `count` ?

Reply:
Done. Refactored the test file a bit as well.